### PR TITLE
feat(docker): Build multiple python version images

### PR DESCRIPTION
**Pull request by Germandrummer92**
### SUMMARY
We would like to use python3.9/3.10 in our custom configuration code to have some nice-to-have features. I don't really see a reason why there shouldn't be a python3.9/python3.10 version, the build arg even already exists?

### TESTING INSTRUCTIONS
Run the docker build and see it works with python3.9/python3.10

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [X] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by rusackas** [comment]
Seems related to https://github.com/apache/superset/discussions/22810

File: .github/workflows/docker_build_push.sh [hunk 1]
```diff
@@ -50,12 +50,39 @@ docker build --target lean \
...
 
+# Build the "lean39" image
+docker build --target lean \
+  -t "${REPO_NAME}:${SHA}-py39" \
+  -t "${REPO_NAME}:${REFSPEC}-py39" \
+  -t "${REPO_NAME}:${LATEST_TAG}-py39" \
+  --build-arg PY_VER="3.9.16"\
+  --label "sha=${SHA}" \
+  --label "built_at=$(date)" \
+  --label "target=lean39" \
```
Comment (craig-rueda) [inline .github/workflows/docker_build_push.sh:68]: I don't think this target exists in the `Dockerfile`

File: .github/workflows/docker_build_push.sh [hunk 2]
```diff
@@ -50,12 +50,39 @@ docker build --target lean \
...
+  --label "build_actor=${GITHUB_ACTOR}" \
+  .
+
+
+# Build the "lean310" image
+docker build --target lean \
```
Comment (craig-rueda) [inline .github/workflows/docker_build_push.sh:74]: Let's only go with `3.9` for now as `3.10` isn't tested yet

Comment (Germandrummer92) [reply .github/workflows/docker_build_push.sh]: removed 3.10

**Comment by Germandrummer92** [comment]
@rusackas I removed the patch version now and switched back to the -slim version. Let me know if that makes sense or if we should stay with the pinned patch version. 

I think -slim for sure makes sense if it works, which it did before.

**Comment by Germandrummer92** [comment]
Maybe there also should be a defined process of when the "next-python" image becomes the "main" python version? 

python3.8 is still supported until next year, but in general I wouldn't mind if we only lag about 1 year behind the python release schedule? https://endoflife.date/python

That would mean we would try to move the "main" python image to 3.9 and have 3.10 support soon; and move to 3.10/3.11 in October when 3.12 releases?

**Review by craig-rueda — APPROVED** [review]

**Pull request merged**